api/routes/user.py

```diff
@@ -18,3 +21,59 @@ def get_current_user_profile(
 
     log(log.INFO, f"User {current_user.username} requested his profile")
     return current_user
+
+
+@user_router.patch("/me/language", status_code=status.HTTP_200_OK, response_model=s.User)
+def update_user_language(
+    language_update: s.LanguageUpdate,
+    current_user: m.User = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """Updates the current user's language"""
+    if current_user.language != language_update.language:
+        current_user.language = language_update.language
+
+    db.add(current_user)
+    db.commit()
+    db.refresh(current_user)
+
+    log(log.INFO, f"User {current_user.username} updated his language to {current_user.language}")
+    return current_user
+
+
+@user_router.patch("/me/username", status_code=status.HTTP_200_OK, response_model=s.User)
+def update_user_username(
+    username_update: s.UsernameUpdate,
+    current_user: m.User = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """Updates the current user's username"""
+    if current_user.username != username_update.username:
+        current_user.username = username_update.username
+
+    db.add(current_user)
+    db.commit()
+    db.refresh(current_user)
+
+    log(log.INFO, f"User {current_user.username} updated his username to {current_user.username}")
+    return current_user

```

Comment:
FYI, I don't see point in creating a separate endpoint to update each param of user. You may consider having single PATCH method, which will accept all params you wish to change as optional, and then update only those provided.

Reply:
Also, regarding username update. Is username used for users to login? if so, then I think it would be a good idea NOT!!!!! to let users change their usernames. We need to have a separate field called smth like `display_name` and it will be used at UI and user can change this value